### DISCUSSION TITLE
TasmoManager - A modern, sleek and responsive UI for managing Tasmota devices

### DISCUSSION THREAD

**Discussion by tom-23** [arendst/Tasmota#11509] [Show and tell]
Hi There,

I've been developing this app for managing Tasmota devices for about a month now. I could really do with some feedback / potential ideas on how to improve it and make it better! One of the main purposes of creating this project was to try my hand at writing production-ready, Windows and macOS apps plus making easy to use and user-friendly GUIs. I've already implemented features like Bulk Firmware Updating, Integrated Terminal, SetOptions, Device Control and a Setup Wizard which finds new Tasmota devices on the network which haven't connected to MQTT yet. This project was heavily inspired by TDM and TasmoAdmin (both amazing projects btw).

<img width="1169" alt="MainWindow" src="https://user-images.githubusercontent.com/24756018/112864088-67cb8600-90af-11eb-8147-0eefe33f60b7.png">

Let me know your thoughts!

https://github.com/tom-23/TasmoManager/releases

**Comment by sfromis** [comment]
One practical detail. I'd expect Linux to be "more common than average" among Tasmota users, and this platform not being available does reduce the audience.

**Reply by tom-23** [reply]
Fair shout! In theory, Linux support should be really easy to support. In fact, I could probably build it without any issues! I'm gonna look into some Linux stuff and see how easy it is to make a Debian package or something.

**Reply by tom-23** [reply]
@sfromis I've just added Linux support (in the form of Linux packages). https://github.com/tom-23/TasmoManager/releases/tag/v1.0.0-alpha.11

**Reply by sfromis** [reply]
After installing the deb on Ubuntu 20.04 (near pristine state), trying to run failed silently. Doing it in a terminal window gave 
`error while loading shared libraries: libQt5Qmqtt.so.1: cannot open shared object file: No such file or directory`
While I could hunt down what needs to be installed, I'd expect the install to take care of prerequisites.

**Reply by tom-23** [reply]
I thought cqtdeployer would copy over this library but *apparently* it didn't. All fixed in [1.0.0-alpha.12](https://github.com/tom-23/TasmoManager/releases/tag/v1.0.0-alpha.12). I've tested this on a blank Ubuntu 20.04 VM and everything seemed to work fine so fingers crossed it'll run for you.

**Reply by TukkerManS** [reply]
I see some strange things after updating 3 Tasmotas (1x 6.4.1 2x 6.5.0) to 10.0.0 I don't know if it is an error so I will report it here
After updating, 192.168.178.150, 151 and 152 have the same name and I cannot change it. Only the first one. What to do.
And I miss the option to delete unused Tasmotas.

**Reply by barbudor** [reply]
I don't believe this is related to TasmoManager but many people who upgraded from old version to ~v10 falled into a constraint in this Tasmota version where Topic and MqttClient has to be different. Thus constraint has been removed in dev version 10.1.0.1
So this is available in release 11.0

**Reply by TukkerManS** [reply]
Yep, you have right. I made a mistake. i upgraded 3 Tasmotas (1x 6.4.1 2x 6.5.0) to 11.0.0
![Foto-2](https://user-images.githubusercontent.com/44361858/160164483-b92ba8f8-dd99-4952-8e97-7b5e375ae669.jpg)
![Foto-3](https://user-images.githubusercontent.com/44361858/160164497-56fdf3f2-1b2c-46cf-bd7f-12e32cb55e7f.jpg)
![Foto-4](https://user-images.githubusercontent.com/44361858/160164507-678bdacd-258f-4b84-8751-3f9406dc8b28.jpg)

**Comment by arendst** [comment]
Looks nice! Works on windows ;-)

Some issues:
- I do see a lot of ``Tasmota Device`` without possibility to open the GUI
- I see duplicates as ``Tasmota Device`` and ``Device name`` (See IP address 192.168.2.211 and .225)

![image](https://user-images.githubusercontent.com/11044339/112867540-57210d00-90bb-11eb-9b9e-f77423e1bd9b.png)

I like the MQTT comms over HTTP comms!

BTW .225 is a different device than Wemos10. Clicking on it closes the windows (probably in error). Same happens when I select a line and move the mouse up. After some line the window closes.

**Reply by tom-23** [reply]
TasmoManager uses the "tasmota/discovery" topic for device discovery so it is probably finding old devices in that topic. I've just released v1.0.0-alpha.8 which allows the user to hide Offline devices. As for the other issue, I'll have to do some more testing to find out what could be causing this issue. Thanks for checking it out btw!

**Reply by sfromis** [reply]
Using the MAC-keyed /discovery topics may be problematic due to the risk of very old retained topics when devices are swapped for the same topic/function. I'm using /LWT for discovery, but of course that does not provide details. For "best of both", I'd probably discard (or at least flag) /discovery retained messages not matching the topic from the LWT messages. Another issue with using the /discovery topics is that it is a relatively recent feature, not great for scenarios including older Tasmota versions.

**Reply by barbudor** [reply]
What are you meaning with "discovery" ?
`USE_DISCOVERY` was mDNS and removed
Or HASS Discovery ? that I have removed from my bins as I don't have an HASS.

**Reply by sfromis** [reply]
In this context, "discovery" means the `tasmota/discovery/xxxxxxxxxxxx/config` and maybe `.../sensors` retained topics published to MQTT, indeed created along with native Tasmota integration with Home Assistant, very different from the older MQTT Discovery feature of that. With `SetOption19 0` the "new discovery" topics are published, and they should work as well for any other platform wanting easy discovery of metadata - as long as builds still include `USE_HOME_ASSISTANT`. IMO, this should be made a separate selectable feature without any HA ties.

**Reply by barbudor** [reply]
ok, thanks. I'll give it a try.

**Reply by arendst** [reply]
Ah ``tasmota/discovery``. I didn't know I had it... Apparently some HomeAssistant enabled devices did put  retained info in my mqtt server. Just removed them all. Now this tool doesn't work anymore as you said it needs HomeAssitant discovery data. Ok, no big deal but I have to drop off this time.

**Comment by TrDA-hab** [comment]
It is very cool ! 
Especially for macOS (I always miss this). I'm waiting for the final release !!!

**Comment by TrDA-hab** [comment]
Now I installed the latest version, I see only 20% of my devices, 50% of the declared functions do not work or do not work correctly. This is a good start!
![Снимок экрана 2021-04-13 в 11 38 12](https://user-images.githubusercontent.com/56403720/114523872-66709080-9c4d-11eb-83a9-cdb8d957ad53.png)

**Reply by arendst** [reply]
I had that too until I realized TasmoManager needs the new Hass discovery data for populating it's screen.

As the current HASS discovery code is over 12k I extracted and updated the new Hass Discovery code into a new driver called Tasmota Discovery which is almost 2k in size. With that enabled all my devices pop up in both TasmoManager AND Hass!

**Reply by TrDA-hab** [reply]
I am currently using another smart home server MajorDoMo, but MQTT is also used there Mosquitto broker and Tasmota Device Manager works great there !. HA I'm just studying so far.

**Reply by arendst** [reply]
What I tried to say is that you do not need the HASS discovery anymore. Tasmota discovery will do just fine for all home servers which support Tasmota Discovery (not many yet but at least Home Assistant AND TasmoManager work with it).

**Reply by TrDA-hab** [reply]
Great news! I will be switching to Home Assistant soon. Thank you!!!

**Reply by sfromis** [reply]
If you want to have TasmoManager working, you do not need to switch to Home Assistant, just use binaries compiled to include `USE_TASMOTA_DISCOVERY` **or** of course HA.

**Reply by TrDA-hab** [reply]
Thanks for the info!

**Reply by NsinghP** [reply]
> 
> 
> If you want to have TasmoManager working, you do not need to switch to Home Assistant, just use binaries compiled to include `USE_TASMOTA_DISCOVERY` **or** of course HA.

How to do that? Which additional feature to select for Tasmota Discovery when you chose wemos module in TasmoCompiler? I have already compiled with Home Assistant Feature. Do I just need to compile again with #define USE_TASMOTA_DISCOVERY custom parameter change to User_config_override_h file?

**Reply by Jason2866** [reply]
Yes, just put `#define USE_TASMOTA_DISCOVERY` in your `user_config_override.h`

**Reply by digiblur** [reply]
Looks like Home Assistant still picks them up unfortunately with that discovery on instead. I tried to add them manually but it wouldn't find them.

It is a promising app though! Just needs additional ways to add devices. Like also listening to tele/#

**Reply by NsinghP** [reply]
> Yes, just put `#define USE_TASMOTA_DISCOVERY` in your `user_config_override.h`

Compile with that line failed due to following errors:

: undefined reference to `HAssDiscover()'

: undefined reference to `HAssPublishStatus()'

What am I missing?

**Reply by arendst** [reply]
You either select USE_TASMOTA_DISCOVERY or USE_HOME_ASSISTANT. Not both

**Reply by NsinghP** [reply]
> You either select USE_TASMOTA_DISCOVERY or USE_HOME_ASSISTANT. Not both

Ok, Got it, 

I need to //comment out #Define USE_HOME_ASSISTANT in User_config.... file

PS. I guess I cant do that using tasmocompiler!

**Reply by NsinghP** [reply]
> You either select USE_TASMOTA_DISCOVERY or USE_HOME_ASSISTANT. Not both

I can’t figure out how to select between them! There is no option to select Tasmota discovery in tasmocompiler but there is option to select Home Asistant. How do I select USE_Tasmota_Discovery?

**Reply by barbudor** [reply]
There is a section for custom flags
Add
`#define USE_TASMOTA_DISCOVERY `
and do not select Home Assistant

**Reply by NsinghP** [reply]
> There is a section for custom flags
> Add
> `#define USE_TASMOTA_DISCOVERY `
> and do not select Home Assistant

Ok, I have already added #define USE_TASMOTA_DISCOVERY to custom strings and not selected Home Asistant in features section. Bin file compiles without error. But my devices are still not visible on Tasmo Manager! I guess it must be something else-like not having MQTT configured.

PS: Added tasmomanger as an app allowed through windows firewall. Devices still not visible.

**Comment by digiblur** [comment]
Is there a portable windows version available?  I could see several people requesting a simple unzip into a folder and run method.

**Reply by tom-23** [reply]
There is a portable zip, just it doesn't get uploaded to GitHub releases automatically. I've just updated the latest release to have a windows portable zip: https://github.com/tom-23/TasmoManager/releases/tag/v1.0.0-alpha.12 I haven't tested it to see if it works on windows but in theory, it should launch without any issues. Let me know if there are any problems! (btw, I'll push windows portable artifacts in the future :) )

**Comment by NsinghP** [comment]
>  Device Control and a Setup Wizard which finds new Tasmota devices on the network which haven't connected to MQTT yet.

Hi Tom, Your app looks very promising. I have installed it on windows 10, but it is not picking up tasmota devices on my home wifi network. I have 5 devices active on the network. I tried both with automatic method and specify IP range method. These devices are not configured to connect to MQTT.

Regards, 
Navi
PS: I am running Tasmota 9.3.1.3 compiled with Home Assistant Support. Setoption19 is set to 0. MQTT is enabled but not configured. Hue Bridge Emulation is turned on. I can see all devices in windows network as Amazon-Echo-HA-Bridge (192.168.1.xxx)

**Comment by mkormendy** [comment]
I've been using https://github.com/reloxx13/TasmoAdmin for a number of years now. This is certainly a different approach, I downloaded it and will test it out.
